tweetapp/views.py

```diff
@@ -12,29 +12,44 @@
 from django.http import JsonResponse
 # Create your views here.
 
+STOP_WORDS = set(stopwords.words('english'))
+
+
+# cache this so it's only updated every three minutes.
+def _get_trending_topics() -> list[tuple]:
+    # scope our trending tweets to the last 3 hours only
+    datetime_bound = datetime.now() - timedelta(hours=3)
+    # [{ "content": 'tweet content' }, {}]
+    tweet_contents = Tweet.objects.filter(updated__gte=datetime_bound).values_list("content")
 
-def trendingView(request):
-    stop_words = set(stopwords.words('english'))
-    tweets = Tweet.objects.all()
     words = []
-    for tweet in tweets:
+    for tweet_content in tweet_contents:
+        tweet = tweet_content['content']
         # Extract words from post content and filter out stop words
         words.extend([
             word.lower() for word in re.findall(r'\w+', tweet.content)
-            if word.lower() not in stop_words
+            if word.lower() not in STOP_WORDS
         ])
-
         # Extract hashtags from post content
         hashtags = re.findall(r'#\w+', tweet.content)
         words.extend(hashtags)
         word_count = Counter(words)
```

Comment:
For production twitter, we could aggregate the word counts from each tweet as we go in a counter dict, instead of building a massive list of all words and then computing the counts.